### ISSUE TITLE
Becomes unresponsive with large amounts of data.

### ISSUE THREAD

**Issue by tymondeusz** [abdularis/LAN-Share#14]
Hello, 
Great app. I really love it, BUT... 

It becomes unresponsive with large amounts of data. When upgrading my PC and migrating data off my trusty old laptop the app became unresponsive for  a while before starting transfer. I understand it's probably preparing the transaction, but there is no feedback of any kind. App becomes unresponsive, but now I know from experience that when I was moving some 60GB of data it would get stuck for a good few minutes, but eventually would begin transfer (and once started it's blazing fast!). 

Now I'm trying to move 1.7TB of data and it's been unresponsive for the past hour or more, lol. 

One curious thing I noticed is that it is fully utilising one of the CPU threads and it swaps from one to another. It will put 100% load on thread one, then move to thread 2, to thread 3 and so on, but it's been doing that for an hour. May be beneficial if the app were to use all CPU cores / threads maybe and give a sort of feedback as to what is going on in the background. I have no way of telling if it's actively working on  preparing the transaction or it's just GOOOOONE :D 

My only clue is that the CPU load transfers from one thread to another with varying frequency. Sometimes sticks for a few minutes, sometimes will jump from thread to thread within seconds.

**Comment by abdularis** [comment]
Yeah, when we have large number of files from folder for example, the application will try to iterate all files contained in that folder (recursively) in main thread, and we haven't add any progress or something to tell the user, :D
trying to improve...

**Comment by Gabriel-p** [comment]
Same issue here with less than 3Gb of data Ubuntu to Ubuntu. Showing a progress bar for the entire process would help a lot, otherwise it looks like the app just hanged.

Great tool BTW!

**Comment by rwperrott** [comment]
There should be a config. value to specify the maximum number of active file transfers, to prevent the idiocy of trying to transfer all, of many, files at once.
